program/src/processor.rs

```diff
@@ -231,6 +260,8 @@ fn move_stake_or_lamports_shared_checks(
     }
 
     // source and destination must be writable
+    // JC nit: we can probably remove these too since the runtime will definitely
+    // catch this
     if !source_stake_account_info.is_writable || !destination_stake_account_info.is_writable {
         return Err(ProgramError::InvalidInstructionData);
     }
```

Comment:
this is because move stake and move lamports are defined by simd to abort if theyre not writable and i didnt want to worry about there being any possible codepath that resulted in no attempt to write and a successful no-op. but if we are going to have one impl for all validator clients it matters less

Reply:
Ah ok, if it's in the SIMD, let's leave as is